optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py

```diff
@@ -0,0 +1,717 @@
+        second element is the number of inference steps.
+    """
+    if timesteps is not None:
+        accepts_timesteps = "timesteps" in set(inspect.signature(scheduler.set_timesteps).parameters.keys())
+        if not accepts_timesteps:
+            raise ValueError(
+                f"The current scheduler class {scheduler.__class__}'s `set_timesteps` does not support custom"
+                f" timestep schedules. Please check whether you are using the correct scheduler."
+            )
+        scheduler.set_timesteps(timesteps=timesteps, device="cpu", **kwargs)
+        timesteps = scheduler.timesteps.to(device)
+        num_inference_steps = len(timesteps)
+    else:
+        scheduler.set_timesteps(num_inference_steps, device="cpu", **kwargs)
+        timesteps = scheduler.timesteps.to(device)
+
+    # Handles the case where the scheduler cannot implement reset_timestep_dependent_params()
+    # Example: UniPCMultiStepScheduler used for inference in ControlNet training as it has non-linear accesses to timestep dependent parameter: sigma.
+    if hasattr(scheduler, "reset_timestep_dependent_params") and callable(scheduler.reset_timestep_dependent_params):
+        scheduler.reset_timestep_dependent_params()
+    return timesteps, num_inference_steps
+
+
+class GaudiStableDiffusionImg2ImgPipeline(GaudiDiffusionPipeline, StableDiffusionImg2ImgPipeline):
+    """
+    Adapted from: https://github.com/huggingface/diffusers/blob/v0.26.3/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L161
+    Changes:
+        1. Use CPU to generate random tensor
+    """
+
+    def __init__(
+        self,
+        vae: AutoencoderKL,
+        text_encoder: CLIPTextModel,
+        tokenizer: CLIPTokenizer,
+        unet: UNet2DConditionModel,
+        scheduler: KarrasDiffusionSchedulers,
+        safety_checker: StableDiffusionSafetyChecker,
+        feature_extractor: CLIPImageProcessor,
+        image_encoder: CLIPVisionModelWithProjection = None,
+        requires_safety_checker: bool = True,
+        use_habana: bool = False,
+        use_hpu_graphs: bool = False,
+        gaudi_config: Union[str, GaudiConfig] = None,
+        bf16_full_eval: bool = False,
+    ):
+        GaudiDiffusionPipeline.__init__(
+            self,
+            use_habana,
+            use_hpu_graphs,
+            gaudi_config,
+            bf16_full_eval,
+        )
+
+        StableDiffusionImg2ImgPipeline.__init__(
+            self,
+            vae,
+            text_encoder,
+            tokenizer,
+            unet,
+            scheduler,
+            safety_checker,
+            feature_extractor,
+            image_encoder,
+            requires_safety_checker,
+        )
+        self.to(self._device)
+
+    # Copied from ./pipeline_stable_diffusion.py
+    @classmethod
+    def _split_inputs_into_batches(cls, batch_size, latents, prompt_embeds, negative_prompt_embeds):
```

Comment:
Lower num_prompts_per_image is throwing error- could be we need some fix or input validation:
```
python image_to_image_generation.py \
    --model_name_or_path "ogkalu/Comic-Diffusion"\
    --src_image_path "
[https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg"](https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg%22)
\
    --prompts "turn him into cyborg" \
    --num_images_per_prompt 10 \
    --batch_size 4 \
    --guidance_scale 7.5 \
    --image_guidance_scale 1 \
    --num_inference_steps 10 \
    --image_save_dir stable_diffusion_images \
    --use_habana \
    --use_hpu_graphs \
    --gaudi_config Habana/stable-diffusion \
    --bf16
```
```
Traceback (most recent call last):
  File "/root/PR/pi314ever_oh/examples/stable-diffusion/image_to_image_generation.py", line 338, in <module>
    main()
  File "/root/PR/pi314ever_oh/examples/stable-diffusion/image_to_image_generation.py", line 300, in main
    outputs = pipeline(
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py", line 462, in __call__
    latents_batches, text_embeddings_batches, num_dummy_samples = self._split_inputs_into_batches(
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py", line 180, in _split_inputs_into_batches
    prompt_embeds_batches = torch.stack(prompt_embeds_batches)
RuntimeError: stack expects each tensor to be equal size, but got [8, 77, 768] at entry 0 and [4, 77, 768] at entry 2
```

Reply:
Was caused by incorrect usage of `batch_size`, which was assumed to be `num_prompts` in the original. This was corrected.